src/accelerate/optimizer.py

```diff
@@ -161,7 +165,26 @@ def step_was_skipped(self):
         return self._is_overflow
 
     def __getstate__(self):
-        return self.__dict__.copy()
+        ret = {}
+        for k, v in self.__dict__.items():
+            if k not in [
+                "_accelerate_num_step_called",
+                "_optimizer_original_step_method",
+                "_optimizer_patched_step_method",
+            ]:
+                ret[k] = v
+        return ret
 
     def __setstate__(self, state):
         self.__dict__.update(state)
+        self._accelerate_num_step_called = 0
+        self._optimizer_original_step_method = self.optimizer.step
+        self._optimizer_patched_step_method = patch_optimizer_step(self, self.optimizer.step)
+
+
+def patch_optimizer_step(accelerated_optimizer: "AcceleratedOptimizer", method):
```

Comment:
We can use the class directly for typehints

```suggestion
def patch_optimizer_step(accelerated_optimizer: AcceleratedOptimizer, method):
```

Reply:
I update this in the latest commit